joomla/cassiopeia
# [Metismenu] [Draft] Rename metismenu layout and add minified js

Pull Request for Issues [https://github.com/joomla/cassiopeia/pull/195#issuecomment-715308105](https://github.com/joomla/cassiopeia/pull/195#issuecomment-715308105), #208 and [https://github.com/joomla/cassiopeia/pull/210#issuecomment-718710233](https://github.com/joomla/cassiopeia/pull/210#issuecomment-718710233).

Alternative to PR's #217 and #221 .

### Summary of Changes

This pull request (PR) changes the name of the layout to something more meaningful than "metismenu" and adds a language string for a nice layout name for the layout selection in the menu module's advanced options.

The js file which had es6 syntax and a plain`.js` suffix was changed to es5 syntax so it fits to that suffix, and a minified file is added and loaded if present.

Finally, missing doc block with copyright information has been added to the js file.

### Testing Instructions

If you have development environment for Joomla 4, checkout the [development-rename-metismenu-layout-and-add-minified-js](https://github.com/joomla/cassiopeia/tree/development-rename-metismenu-layout-and-add-minified-js) branch, pull the latest changes, run "composer install" and "npm ci", and then make a new installation.

If you don't have that, use the following package to make a new installation: (obsolete link removed)

After that, login to backend and install "Blog Sample Data" to see the metismenu being used in frontend.

Then, still in backend, check the available layouts in the advanced options of the "Main Menu Blog" module.

Last but not least, go to frontend and verify that the metismenu works as well as before. You can also duplicate it in backend to another module position, or create another metismenu, and check in frontend that those modules still work well.

Finally, verify that file `templates/cassiopeia/js/mod_menu/menu-metismenu.js ` has es5 and not es6 syntax and a doc block. Verify that there is a minified file and that the minified file is loaded when debug is off.

### Actual result BEFORE applying this Pull Request

Layout selection in the menu module's advanced options:

![2020-10-29_02](https://user-images.githubusercontent.com/7413183/97583428-3ca51380-19f7-11eb-914f-3f8419313f46.png)

The javascript file `templates/cassiopeia/js/mod_menu/menu-metismenu.js` to load the metismenus has es6 syntax but just a plain "js" file name extension. There is no minified file of it.

### Expected result AFTER applying this Pull Request

Layout selection in the menu module's advanced options:

![2020-11-03_02](https://user-images.githubusercontent.com/7413183/97963690-d2ee8600-1db7-11eb-9cb1-003d37662690.png)

The metismenu (one or more than one at a time) works as well as without this PR.

The  javascript file `templates/cassiopeia/js/mod_menu/menu-metismenu.js ` to load the metismenu has es5 syntax and a doc block. There is a minified version, too, which is loaded when debug is off.

### Documentation Changes Required

The alternative layout should be described in the J4 documentation for the Cassiopeia template.
